### PR TITLE
refactor: Add support for evm chain configuration

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Implements https://app.clickup.com/t/9002019994/PRF-4990

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4990 Add support for EVM chains](https://app.clickup.com/t/9002019994/PRF-4990)

**Review by kartikCypherock — APPROVED** [review]
LGTM

File: apps/evm_family/evm_main.h [hunk 1]
```diff
@@ -44,6 +45,6 @@
  *
  * @param usb_evt The USB event which triggered invocation of the EVM app
  */
```
Comment (amanCypherock) [inline apps/evm_family/evm_main.h:47]: Doxygen comment update required

File: apps/evm_family/evm_context.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+/*****************************************************************************
+ * TYPEDEFS
+ *****************************************************************************/
+
+typedef struct {
+  /** Largest unit/denomination indicator/symbol. This will be used in UX for
+   * displaying fees and amount.
+   */
+  const char lunit_name[EVM_SHORT_NAME_MAX_SIZE];
```
Comment (amanCypherock) [inline apps/evm_family/evm_context.h:38]: Would we also need the lunit conversion decimal value in this config?

Comment (ujjwal-cyph) [reply apps/evm_family/evm_context.h:38]: no. did not find it necessary. all the currently supported chains have the same decimal value (18). the decimal value varies in tokens only. ref: https://app.clickup.com/t/9002019994/PRF-4320

File: apps/evm_family/evm_main.h (hunk 1 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_main.h:49]: added

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**